### DISCUSSION TITLE
Use a second core to control peripherals

### DISCUSSION THREAD

**Discussion by MrCorsair3** [micropython/micropython#13747] [ESP32]
Hello,
I have one project where i have to control a lot of ws2815 rgb leds (~500-600 per one pin).
To control this LEDs I use arduino cloud micropython library and dashboard on the web page.
Generally it works for small count of LEDs but l have a problem with large count of them.
I use integrated library NeoPixel, but control LEDs is blocking operation (bitstream send) and is not possible to use e.g. asyncio to work around this problem.
I have an idea to use second core in ESP32 to only control LEDs. Main program will be run on the first core and they can be communicate between each other by e.g. share memory field.
But I haven't found find any information how I can to this. Generally i think to use micropython with aruduino cloud library on the first core and control program to drive led  on second core (can be writte in C++ example by xTaskCreatePinnedToCore).
I dont know how to make project like this, how install pure Arduino at second core and micropython for first core.

Maybe exist simple way, but i dont see them.

Does anyone have an idea how to solve this problem?

Regards

**Comment by rkompass** [comment]
> I dont know how to make project like this, how install pure Arduino at second core and micropython for first core.

As far as I know this has not been done yet. It certainly is not a simple problem.
